### DISCUSSION TITLE
Filament Tracking Improvements

### DISCUSSION THREAD

**Discussion by joeycortez42** [OctoFarm/OctoFarm#1325] [Feature Request]
### OctoFarm Version

1.7.3

### OctoPrint Version(s)

1.8.6

### pm2 or docker?

docker

### Operating System

Mac OS

### Description of the problem

I am trying to migrate away from the Filament Manager plugin since support has been dropped. But I noticed some issues with using OctoFarms built-in filament tracking.

- [ ] List View: Tool 0: Alphabetical sorting of spools in dropdown. Currently sorts by creation index.
- [ ] Filament View: Alphabetical sorting of spools. Currently sorts by creation index.
- [ ] Filament View: Profile Manager: Alphabetical sorting of profiles. Currently sorts by creation index.
- [ ] Filament View: Spool Manager: Rename "Used (g)" column to "Remaining (g)"
- [ ] Filament View: Spool Manager: Changes to spool data like "Used (g)" field not always saved. Sometimes duplicate sets of spools appear in the view but are gone after a page refresh.
- [ ] Filament View: Spool data is inaccurate or miscalculated. Data seems like some spools are calculated by filament used and others are calculated by filament remaining.

### Steps taken to reproduce

Use OctoFarm Filament tracking.

[octofarm_dump-10-19-2022-9-17-05 AM.zip](https://github.com/OctoFarm/OctoFarm/files/9820793/octofarm_dump-10-19-2022-9-17-05.AM.zip)

![Screen Shot 2022-10-19 at 9 18 19 AM](https://user-images.githubusercontent.com/1263913/196702152-cc0f64a9-4e2f-4f8c-8fed-2111b70b302c.png)

**Comment by NotExpectedYet** [comment]
Hello, I've moved this to discussions because that's where my feature requests live. 

My patreon users have asked for my focus for the next 3 month of development to be history and filament manager so I'll take a look at adding these in. Thanks for the suggestions. 

Bugs will need issues creating for them so they can be tracked on sorted.

**Reply by orangeworx** [reply]
I've arrived here rather unexpectedly as I was looking for something else but OP's post is very close to what I've come to experience and most recently as I was binge listing my spools.
Happy to hear filament manager is a priority.  I'm very curious as to how that's going to advance.
I also don't mind testing stuff :D

Keep it up! Awesome stuff!

**Reply by NotExpectedYet** [reply]
> I've arrived here rather unexpectedly as I was looking for something else but OP's post is very close to what I've come to experience and most recently as I was binge listing my spools. Happy to hear filament manager is a priority. I'm very curious as to how that's going to advance. I also don't mind testing stuff :D
> 
> Keep it up! Awesome stuff!

I'm currently buying a house so free time has taken a hit but planning to get back on it soon. 

Could do with more testers tbh! It's not something easy to setup currently but I will hopefully have a development versions and also rc versions configurable to update from the UI in the next release. The UI will also show the current and future changelogs. 

I think the entire view could do with a decent re-working. 

If anyone wants to take a stab with a PR or fire more wants my way they are welcome.

**Reply by orangeworx** [reply]
Real life takes precedence! Good luck with that!
if you ever make a list for potential tester, sign me up :)

Looking forward!
